airflow/utils/scheduler_health.py

```diff
@@ -36,11 +38,11 @@ def do_GET(self):
             try:
                 with create_session() as session:
                     scheduler_job = (
-                        session.query(Job)
+                        session.scalars(select(Job))
                         .filter_by(job_type=SchedulerJobRunner.job_type)
                         .filter_by(hostname=get_hostname())
                         .order_by(Job.latest_heartbeat.desc())
-                        .first()
+                        .limit(1)
                     )
```

Comment:
This change is wrong!

Reply:
sorry, why is this wrong @uranusjr ?

Reply:
You’re calling `filter_by` etc. against the result of `session.scalars` instead of `select`. The parenthesis on line 41 is probably misplaced?

Reply:
I didnt get any error syntactically or when executing the tests. 🤷‍♂️ 

```
abc=session.query(Job).filter_by(job_type=SchedulerJobRunner.job_type).filter_by(hostname=get_hostname()).order_by(Job.latest_heartbeat.desc()).first()

(Pdb) abc
<MagicMock name='create_session().__enter__().query().filter_by().filter_by().order_by().first()' id='281473176118176'>

def1 = session.scalars(select(Job)).filter_by(job_type=SchedulerJobRunner.job_type).filter_by(hostname=get_hostname()).order_by(Job.latest_heartbeat.desc()).limit(1)

(Pdb) def1
<MagicMock name='create_session().__enter__().scalars().filter_by().filter_by().order_by().limit()' id='281473183539744'>
```

Reply:
Because you mocked the session object; with mock you can literally call anything without errors. This fails if you use an actual SQLAlchemy session object

```pycon
>>> from airflow.utils.db import create_session
>>> from airflow.models.job import Job
>>> from sqlalchemy import select
>>> with create_session() as s:
...  s.scalars(select(Job)).filter_by(job_type="SchedulerJob")
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AttributeError: 'ScalarResult' object has no attribute 'filter_by'
```